### DISCUSSION TITLE
Processing documents to derive topics and research questions

### DISCUSSION THREAD

**Discussion by danieldekay** [assafelovic/gpt-researcher#788] [Ideas]
I am currently working a lot with meeting transcripts, which I want to write into small reports (and not meeting minutes). It actually works quite well, if I give a longer prompt as research question, but if I just say "Summarize the transcript and write a report"  GPTR starts with building queries like:

![image](https://github.com/user-attachments/assets/063896bc-6bf2-4ff7-afb4-522e732c644f)

It would be really cool, if GPT-R could actually read through the document first, and extract the topics talked about, and from that point on start developing queries for the RAG retrieval and chapter writing.

How would one go about with GPT-R with such a task?

- create a new report format?
- use multi-agents, somehow?

**Comment by ElishaKay** [comment]
Sup @danieldekay 

Maybe the outline report type can help?

Have a look here at [generate_outline_report_prompt](https://github.com/assafelovic/gpt-researcher/blob/master/gpt_researcher/orchestrator/prompts.py)

**Reply by andrepatt** [reply]
Link is broken 🥲

**Reply by ElishaKay** [reply]
Fixed now

**Comment by ElishaKay** [comment]
Sup @danieldekay 

Did you make progress on it?

It sounds like a good use case for a new Langgraph if you want to dig into the framework.

But good idea to think through the design pattern first.

We created a new channel on the discord for managing context: 

https://discord.com/channels/1127851779011391548/1292367231778295859/1293794181826875474

@andrepatt, you'll need to join the discord with this link below for the link above to work

https://discord.com/invite/QgZXvJAccX
